Source/CarthageKit/Archive.swift

```diff
@@ -13,50 +13,50 @@ import ReactiveTask
 
 /// Zips the given input items (recursively) into an archive that will be
 /// located at the given URL.
-public func zipIntoArchive(destinationArchiveURL: NSURL, inputPaths: [String]) -> SignalProducer<(), CarthageError> {
+public func zipIntoArchive(destinationArchiveURL: NSURL, _ inputPaths: [String]) -> SignalProducer<(), CarthageError> {
 	precondition(destinationArchiveURL.fileURL)
 	precondition(!inputPaths.isEmpty)
 
 	let task = TaskDescription(launchPath: "/usr/bin/env", arguments: [ "zip", "-q", "-r", "--symlinks", destinationArchiveURL.path! ] + inputPaths)
 	return launchTask(task)
-		|> mapError { .TaskError($0) }
-		|> then(.empty)
+		.mapError(CarthageError.TaskError)
+		.then(.empty)
 }
 
 /// Unzips the archive at the given file URL, extracting into the given
 /// directory URL (which must already exist).
-public func unzipArchiveToDirectory(fileURL: NSURL, destinationDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
+public func unzipArchiveToDirectory(fileURL: NSURL, _ destinationDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
 	precondition(fileURL.fileURL)
 	precondition(destinationDirectoryURL.fileURL)
 
 	let task = TaskDescription(launchPath: "/usr/bin/env", arguments: [ "unzip", "-qq", "-d", destinationDirectoryURL.path!, fileURL.path! ])
 	return launchTask(task)
-		|> mapError { .TaskError($0) }
-		|> then(.empty)
+		.mapError(CarthageError.TaskError)
+		.then(.empty)
 }
 
 /// Unzips the archive at the given file URL into a temporary directory, then
 /// sends the file URL to that directory.
 public func unzipArchiveToTemporaryDirectory(fileURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
-	return SignalProducer.try {
-			var temporaryDirectoryTemplate: ContiguousArray<CChar> = NSTemporaryDirectory().stringByAppendingPathComponent("carthage-archive.XXXXXX").nulTerminatedUTF8.map { CChar($0) }
+	return SignalProducer.attempt {
+			var temporaryDirectoryTemplate: [CChar] = (NSTemporaryDirectory() as NSString).stringByAppendingPathComponent("carthage-archive.XXXXXX").nulTerminatedUTF8.map { CChar($0) }
```

Comment:
Couldn't because of the error: `Type of expression is ambiguous without more context`. :sob: